articles/getting-started-with-inheritance-using-c/index.md

```diff
@@ -0,0 +1,228 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/getting-started-with-inheritance-using-c/
+title: Getting Started with Inheritance using C#
+description: The basic concepts of
+author:
+date: 2020-11-24T00:00:00-12:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/getting-started-with-inheritance-using-c/hero.jpg
+    alt:  example image
+---
+Inheritance is a unique concept in (OOP) object-oriented programming, it allows the definition of multiple classes using the properties of other classes. We create classes by inheriting the functions and variables of a base class, then we add new functions to enhance its performance.
+<!--more-->
+### Introduction
+Inheritance allows us to maintain, modify, and further improve our code. It provides us the ability to reuse code functionality. Without inheritance, our code will be unstructured, difficult to read, and complex.
+
+A derived class inherits the properties of a base class.
+
+### Table of contents
+1. Base classes vs Derived classes.
+
+2. Protected vs Internal members.
+
+3. The relationship between classes.
+
+4. How constructors and destructors get executed.
+
+
+### Prerequisites
+Before we begin this tutorial, it is recommended the reader have the following:
+
+- A basic understanding of C# programming language.
+
+- A basic understanding of how Classes work.
+
+### Base classes vs Derived classes
+A **Base** class produces derived classes that inherit the properties of the base class.
+
+Inheritance forms a tree-like Hierarchy:
+
+![inheritance tree](/engineering-education/Getting-started-with-inheritance-using-c#/tree-of-inheritance.png)
+
+**In inheritance**, an object is often the object of another class and all derived classes are objects of their base class, also note that constructors are not inherited.
+
+This is how you would specify a class child derived from class father: `class child: father`
+
+### Protected members vs Internal members
+A derived class can access the **protected members** of a base class, on the other hand
+**Internal members** can only be accessed by a declared class within the same domain.
+
+### The proper way to use inheritance
+Before we dig into the code it's important to note that we include the word **Override** when we need to reuse a method in a derived class inherited from a base class.
+
+If we are going to override a method in a derived class we can use the word **Virtual** before the method in a base class.
+
+Below we have an example where we calculate the area and perimeter of a circle using inheritance.
+
+In this piece of code, we'll define two private point coordinates, then we'll define a constructor that takes two integers from the main.
+
+**Note** that the Output statements use a reference (This) to implicitly call the ToString method.
+
+```c#
```

Comment:
I am getting an End-of-file error with this particular code snippet. I suggest you take a look at it. Also, It would be better if you include the corresponding outputs—that way, I, as a reader, know what to expect. Currently, I don't have enough information to comprehend the error and therefore make the changes.

Reply:
**Specify that you have broken down the code into smaller snippets and all these snippets are part of the same codebase.**